### PR TITLE
Yelinskyi

### PR DESCRIPTION
**Pull request by Yelinskyi**

### CODE REVIEW

File: src/main/java/mate/jdbc/Main.java [hunk 1]
```diff
@@ -1,7 +1,42 @@
...
+        manufacturer = new Manufacturer("Rolls Royce", "UK");
+        manufacturerDao.create(manufacturer);
+        // Show all items in the table DB:
+        allManufacturers = manufacturerDao.getAll();
+        allManufacturers.forEach(System.out::println);
+        // Show all items in the table DB:
+        allManufacturers = manufacturerDao.getAll();
+        allManufacturers.forEach(System.out::println);
+        // Is deleted instance in DB:
+        /*manufacturerDao.delete(15L);*/
```
Comment (Max-Pochepets) [inline src/main/java/mate/jdbc/Main.java:31]: why is this commented?

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                PreparedStatement getInstanceManufacturersStatement = connection.prepareStatement(
+                        getManufacturerRequest)) {
+            getInstanceManufacturersStatement.setLong(1, id);
+            ResultSet resultSet = getInstanceManufacturersStatement.executeQuery();
+            if (resultSet.next()) {
+                manufacturer = createManufacturerFromDataDB(resultSet);
+            } else {
+                throw new CustomJdbcException(("The manufacturer with id = "
+                        + id + " doesn't exists!"));
+            }
```
Comment (Max-Pochepets) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:61]: no need to throw this exception

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            return Optional.ofNullable(manufacturer);
+        } catch (SQLException e) {
+            throw new CustomJdbcException("Can't get an instance from DB with id = " + id, e);
```
Comment (Max-Pochepets) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:64]:
```suggestion
            throw new DataProcessingException("Can't get an instance from DB with id = " + id, e);
```
image

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        } catch (NoSuchElementException e) {
+            throw new CustomJdbcException("The instance with id = "
+                    + manufacturer.getId() + " doesn't exists!", e);
+        }
+        Optional<Manufacturer> optional = get(manufacturer.getId());
+        if (optional.isPresent()) {
+            return get(manufacturer.getId()).get();
+        }
+        throw new CustomJdbcException(("The manufacturer with id = "
+                + manufacturer.getId() + " doesn't exists!"));
```
Comment (Max-Pochepets) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:114]: do we need all this?

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    @Override
+    public boolean delete(Long id) {
+        String deleteManufacturerRequest =
+                "UPDATE manufacturers SET is_deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement =
+                        connection.prepareStatement(deleteManufacturerRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setLong(1, id);
+            return createManufacturerStatement.executeUpdate() != 0;
```
Comment (Max-Pochepets) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:126]: can here be a negative value?

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateManufacturerRequest =
+                "UPDATE manufacturers "
+                        + "SET name = ?, country = ? "
+                        + "WHERE id = ? AND is_deleted = false";
+        try {
+            Connection connection = ConnectionUtil.getConnection();
+            PreparedStatement updateManufacturerStatement =
+                    connection.prepareStatement(updateManufacturerRequest);
```
Comment (Max-Pochepets) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:96]: why not use try-with-resources?

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java (hunk 4 above)
Comment (Yelinskyi) [reply src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:114]: I think 'Yes'.
I can return 'manufacturer' but I wouldn't know if the table changes.

But if you have different opinion I'll do it.

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java (hunk 6 above)
Comment (Yelinskyi) [reply src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:96]: I fogot to do it.

**Pull request closed without merging**